sdk/tables/Azure.Data.Tables/MigrationGuide.md

```diff
@@ -0,0 +1,309 @@
+
+```C# Snippet:TablesSample1CreateClient
+// Construct a new <see cref="TableServiceClient" /> using a <see cref="TableSharedKeyCredential" />.
+
+var serviceClient = new TableServiceClient(
+    new Uri(storageUri),
+    new TableSharedKeyCredential(accountName, storageAccountKey));
+```
+
+### Creating a table
+
+Previously in `Microsoft.Azure.Comsmos.Table`, we'd use a `CloudTable` instance to create a table, which is returned from the `CloudTableClient` method on
+`CloudTableClient`.
+
+```C#
+// Create a table client and create the table if it doesn't already exist.
+string tableName = "OfficeSupplies1p1";
+CloudTable table = tableClient.GetTableReference(tableName);
+cloudTable.CreateIfNotExists()
+```
+
+With `Azure.Data.Tables` we have access to all table level operations directly from the `TableServiceClient`. Because the table service client is not affinitzed to
+any one table, it is ideal for scenarios where you need to create, delete, or list more than one cloudTable.
+
+```C# Snippet:TablesSample1CreateTable
+// Create a new table. The TableItem class stores properties of the created table.
+string tableName = "OfficeSupplies1p1";
+TableItem table = serviceClient.CreateTableIfNotExists(tableName);
+Console.WriteLine($"The created table's name is {table.Name}.");
+```
+
+If your intention is to work only in the context of a single table, it's also possible to create a table from the `TableClient`.
+```C# Snippet:TablesMigrationCreateTableWithClient
+// Get a reference to the TableClient from the service client instance.
+var tableClient = serviceClient.GetTableClient(tableName);
+
+// Create the table if it doesn't exist.
+tableClient.CreateIfNotExists();
+```
+
+### Adding data to the table
+
+Let's define an office supply entity so that we can add it to the cloudTable. First we need to define our custom entity types.
+
+Previously in `Microsoft.Azure.Comsmos.Table`, our entity will inherit from the `TableEntity` base class and look like this:
+
+```c#
+public class OfficeSupplyOld : Microsoft.Azure.Cosmos.Table.TableEntity
+{
+	public string Product { get; set; }
+	public double Price { get; set; }
+	public int Quantity { get; set; }
+}
+```
+
+Now in `Azure.Data.Tables`,we will implement the `ITableEntity` interface to define our entity
+
+```C# Snippet:TablesSample2DefineStronglyTypedEntity
+// Define a strongly typed entity by extending the <see cref="ITableEntity"> class.
+public class OfficeSupplyEntity : ITableEntity
+{
+    public string Product { get; set; }
+    public double Price { get; set; }
+    public int Quantity { get; set; }
+    public string PartitionKey { get; set; }
+    public string RowKey { get; set; }
+    public DateTimeOffset? Timestamp { get; set; }
+    public ETag ETag { get; set; }
+}
+```
+
+Now let's populate an entity to add to the table for each version of the client.
+
+For `Microsoft.Azure.Cosmos.cloudTable.TableEntity`:
+
+```c#
```

Comment:
The nittest of nits:   Should we consider normalizing to `C#` for the code fences?   We're currently using `c#` for non-snippets and `C#` for snippets.